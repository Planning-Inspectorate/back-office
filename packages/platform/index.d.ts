import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
