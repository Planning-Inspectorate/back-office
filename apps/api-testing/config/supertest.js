// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
