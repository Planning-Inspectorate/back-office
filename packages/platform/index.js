export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
