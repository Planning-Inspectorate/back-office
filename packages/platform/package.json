{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
