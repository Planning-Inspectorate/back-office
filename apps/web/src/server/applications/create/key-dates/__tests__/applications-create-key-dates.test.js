import { parseHtml } from '@pins/platform';
import supertest from 'supertest';
import { createTestApplication } from '../../../../../../testing/index.js';

const { app, installMockApi, teardown } = createTestApplication();
const request = supertest(app);

describe('Applications create key dates', () => {
	beforeEach(installMockApi);
	afterEach(teardown);

	beforeEach(async () => {
		await request.get('/applications-service/case-officer');
	});

	const baseUrl = '/applications-service/create-new-case/123/key-dates';

	describe('GET /key-dates', () => {
		it('should render the page', async () => {
			const response = await request.get(baseUrl);
			const element = parseHtml(response.text);

			expect(element.innerHTML).toMatchSnapshot();
			expect(element.innerHTML).toContain('Save and continue');
		});
	});

	describe('POST /key-dates', () => {
		describe('Web-side validation:', () => {
			it('should show validation errors if dates are not valid', async () => {
				const response = await request.post(baseUrl).send({
					firstNotifiedDay: 'not_a_number',
					submissionDay: 32
				});
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('Enter a valid day for the first notified date');
				expect(element.innerHTML).toContain('Enter a valid day for the submission date');
			});

			it('should show validation errors if submission date is not in the future', async () => {
				const response = await request.post(baseUrl).send({
					submissionDay: '14',
					submissionMonth: '7',
					submissionYear: '1789'
				});
				const element = parseHtml(response.text);

				expect(element.innerHTML).toMatchSnapshot();
				expect(element.innerHTML).toContain('The submission date should be in the future');
			});
		});
	});
});
