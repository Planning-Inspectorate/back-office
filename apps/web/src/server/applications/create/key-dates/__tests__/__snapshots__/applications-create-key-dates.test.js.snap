// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Applications create key dates GET /key-dates should render the page 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <h2 class=\\"govuk-heading-l\\"> What are the key dates of the case?</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicationName\\">Date first notified of the case</label>
            <div id=\\"applicationName-hint\\"
            class=\\"govuk-hint\\">for example, 27 3 2009</div>
            <div class=\\"govuk-grid-column-one-third\\">
                <div class=\\"govuk-grid-row pins-date-form-group\\">
                    <div class=\\"govuk-form-group\\">
                        <div id=\\"firstNotifiedDay-hint\\" class=\\"govuk-hint\\">Day</div>
                        <input class=\\"govuk-input\\" id=\\"firstNotifiedDay\\" name=\\"firstNotifiedDay\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedDay-hint\\">
                    </div>
                    <div class=\\"govuk-form-group\\">
                        <div id=\\"firstNotifiedMonth-hint\\" class=\\"govuk-hint\\">Month</div>
                        <input class=\\"govuk-input\\" id=\\"firstNotifiedMonth\\" name=\\"firstNotifiedMonth\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedMonth-hint\\">
                    </div>
                    <div class=\\"govuk-form-group\\">
                        <div id=\\"firstNotifiedYear-hint\\" class=\\"govuk-hint\\">Year</div>
                        <input class=\\"govuk-input\\" id=\\"firstNotifiedYear\\" name=\\"firstNotifiedYear\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedYear-hint\\">
                    </div>
                </div>
            </div>
        </div>
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicationName\\">Anticipated submission date</label>
            <div id=\\"applicationName-hint\\" class=\\"govuk-hint\\">for example, 27 3 2023</div>
            <div class=\\"govuk-grid-column-one-third\\">
                <div class=\\"govuk-grid-row pins-date-form-group\\">
                    <div class=\\"govuk-form-group\\">
                        <div id=\\"submissionDay-hint\\" class=\\"govuk-hint\\">Day</div>
                        <input class=\\"govuk-input\\" id=\\"submissionDay\\" name=\\"submissionDay\\"
                        type=\\"text\\" aria-describedby=\\"submissionDay-hint\\">
                    </div>
                    <div class=\\"govuk-form-group\\">
                        <div id=\\"submissionMonth-hint\\" class=\\"govuk-hint\\">Month</div>
                        <input class=\\"govuk-input\\" id=\\"submissionMonth\\" name=\\"submissionMonth\\"
                        type=\\"text\\" aria-describedby=\\"submissionMonth-hint\\">
                    </div>
                    <div class=\\"govuk-form-group\\">
                        <div id=\\"submissionYear-hint\\" class=\\"govuk-hint\\">Year</div>
                        <input class=\\"govuk-input\\" id=\\"submissionYear\\" name=\\"submissionYear\\"
                        type=\\"text\\" aria-describedby=\\"submissionYear-hint\\">
                    </div>
                </div>
            </div>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/123/applicant-information-types/\\">Back</a>
                <button                 class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save and continue</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`Applications create key dates POST /key-dates Web-side validation: should show validation errors if dates are not valid 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <div class=\\"govuk-error-summary\\" aria-labelledby=\\"error-summary-title\\"
            role=\\"alert\\" data-module=\\"govuk-error-summary\\">
                <h2 class=\\"govuk-error-summary__title\\" id=\\"error-summary-title\\"> There is a problem</h2>
                <div class=\\"govuk-error-summary__body\\">
                    <ul class=\\"govuk-list govuk-error-summary__list\\">
                        <li><a href=\\"#keyDates.firstNotifiedDate\\">Enter a valid day for the first notified date</a>
                        </li>
                        <li><a href=\\"#keyDates.submissionDate\\">Enter a valid day for the submission date</a>
                        </li>
                    </ul>
                </div>
            </div>
        </div>
    </div>
    <h2 class=\\"govuk-heading-l\\"> What are the key dates of the case?</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicationName\\">Date first notified of the case</label>
            <div id=\\"applicationName-hint\\"
            class=\\"govuk-hint\\">for example, 27 3 2009</div>
            <div class=\\"govuk-grid-column-one-third\\">
                <div class=\\"govuk-grid-row pins-date-form-group\\">
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"firstNotifiedDay-hint\\" class=\\"govuk-hint\\">Day</div>
                        <p id=\\"firstNotifiedDay-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"firstNotifiedDay\\" name=\\"firstNotifiedDay\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedDay-hint firstNotifiedDay-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"firstNotifiedMonth-hint\\" class=\\"govuk-hint\\">Month</div>
                        <p id=\\"firstNotifiedMonth-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"firstNotifiedMonth\\"
                        name=\\"firstNotifiedMonth\\" type=\\"text\\" aria-describedby=\\"firstNotifiedMonth-hint firstNotifiedMonth-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"firstNotifiedYear-hint\\" class=\\"govuk-hint\\">Year</div>
                        <p id=\\"firstNotifiedYear-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"firstNotifiedYear\\" name=\\"firstNotifiedYear\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedYear-hint firstNotifiedYear-error\\">
                    </div>
                </div>
            </div>
        </div>
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicationName\\">Anticipated submission date</label>
            <div id=\\"applicationName-hint\\" class=\\"govuk-hint\\">for example, 27 3 2023</div>
            <div class=\\"govuk-grid-column-one-third\\">
                <div class=\\"govuk-grid-row pins-date-form-group\\">
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"submissionDay-hint\\" class=\\"govuk-hint\\">Day</div>
                        <p id=\\"submissionDay-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"submissionDay\\" name=\\"submissionDay\\"
                        type=\\"text\\" value=\\"01\\" aria-describedby=\\"submissionDay-hint submissionDay-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"submissionMonth-hint\\" class=\\"govuk-hint\\">Month</div>
                        <p id=\\"submissionMonth-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"submissionMonth\\" name=\\"submissionMonth\\"
                        type=\\"text\\" value=\\"01\\" aria-describedby=\\"submissionMonth-hint submissionMonth-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"submissionYear-hint\\" class=\\"govuk-hint\\">Year</div>
                        <p id=\\"submissionYear-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"submissionYear\\" name=\\"submissionYear\\"
                        type=\\"text\\" value=\\"2032\\" aria-describedby=\\"submissionYear-hint submissionYear-error\\">
                    </div>
                </div>
            </div>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/123/applicant-information-types/\\">Back</a>
                <button                 class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save and continue</button>
            </div>
        </div>
    </form>
</main>"
`;

exports[`Applications create key dates POST /key-dates Web-side validation: should show validation errors if submission date is not in the future 1`] = `
"<main class=\\"govuk-main-wrapper \\" id=\\"main-content\\" role=\\"main\\">
    <div class=\\"govuk-grid-row\\">
        <div class=\\"govuk-grid-column-two-thirds\\">
            <div class=\\"govuk-error-summary\\" aria-labelledby=\\"error-summary-title\\"
            role=\\"alert\\" data-module=\\"govuk-error-summary\\">
                <h2 class=\\"govuk-error-summary__title\\" id=\\"error-summary-title\\"> There is a problem</h2>
                <div class=\\"govuk-error-summary__body\\">
                    <ul class=\\"govuk-list govuk-error-summary__list\\">
                        <li><a href=\\"#keyDates.firstNotifiedDate\\">Enter a valid day for the first notified date</a>
                        </li>
                        <li><a href=\\"#keyDates.submissionDate\\">The submission date should be in the future</a>
                        </li>
                    </ul>
                </div>
            </div>
        </div>
    </div>
    <h2 class=\\"govuk-heading-l\\"> What are the key dates of the case?</h2>
    <form method=\\"post\\" action=\\"\\" novalidate=\\"novalidate\\" class=\\"pins-applications-create\\">
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicationName\\">Date first notified of the case</label>
            <div id=\\"applicationName-hint\\"
            class=\\"govuk-hint\\">for example, 27 3 2009</div>
            <div class=\\"govuk-grid-column-one-third\\">
                <div class=\\"govuk-grid-row pins-date-form-group\\">
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"firstNotifiedDay-hint\\" class=\\"govuk-hint\\">Day</div>
                        <p id=\\"firstNotifiedDay-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"firstNotifiedDay\\" name=\\"firstNotifiedDay\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedDay-hint firstNotifiedDay-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"firstNotifiedMonth-hint\\" class=\\"govuk-hint\\">Month</div>
                        <p id=\\"firstNotifiedMonth-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"firstNotifiedMonth\\"
                        name=\\"firstNotifiedMonth\\" type=\\"text\\" aria-describedby=\\"firstNotifiedMonth-hint firstNotifiedMonth-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"firstNotifiedYear-hint\\" class=\\"govuk-hint\\">Year</div>
                        <p id=\\"firstNotifiedYear-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"firstNotifiedYear\\" name=\\"firstNotifiedYear\\"
                        type=\\"text\\" aria-describedby=\\"firstNotifiedYear-hint firstNotifiedYear-error\\">
                    </div>
                </div>
            </div>
        </div>
        <div class=\\"govuk-form-group\\">
            <label class=\\"govuk-label font-weight--700\\" for=\\"applicationName\\">Anticipated submission date</label>
            <div id=\\"applicationName-hint\\" class=\\"govuk-hint\\">for example, 27 3 2023</div>
            <div class=\\"govuk-grid-column-one-third\\">
                <div class=\\"govuk-grid-row pins-date-form-group\\">
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"submissionDay-hint\\" class=\\"govuk-hint\\">Day</div>
                        <p id=\\"submissionDay-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"submissionDay\\" name=\\"submissionDay\\"
                        type=\\"text\\" aria-describedby=\\"submissionDay-hint submissionDay-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"submissionMonth-hint\\" class=\\"govuk-hint\\">Month</div>
                        <p id=\\"submissionMonth-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"submissionMonth\\" name=\\"submissionMonth\\"
                        type=\\"text\\" aria-describedby=\\"submissionMonth-hint submissionMonth-error\\">
                    </div>
                    <div class=\\"govuk-form-group govuk-form-group--error\\">
                        <div id=\\"submissionYear-hint\\" class=\\"govuk-hint\\">Year</div>
                        <p id=\\"submissionYear-error\\" class=\\"govuk-error-message\\"><span class=\\"govuk-visually-hidden\\">Error:</span>
                        </p>
                        <input class=\\"govuk-input govuk-input--error\\" id=\\"submissionYear\\" name=\\"submissionYear\\"
                        type=\\"text\\" aria-describedby=\\"submissionYear-hint submissionYear-error\\">
                    </div>
                </div>
            </div>
        </div>
        <div class=\\"govuk-grid-row\\">
            <div class=\\"govuk-button-group pins-button-group govuk-grid-column-one-half\\"><a class=\\"govuk-link\\" href=\\"/applications-service/create-new-case/123/applicant-information-types/\\">Back</a>
                <button                 class=\\"govuk-button\\" data-module=\\"govuk-button\\">Save and continue</button>
            </div>
        </div>
    </form>
</main>"
`;
